sympy/matrices/expressions/matexpr.py

```diff
@@ -110,7 +110,11 @@ def __rmul__(self, other):
     def __pow__(self, other):
         if not self.is_square:
             raise ShapeError("Power of non-square matrix %s" % self)
-        if other is S.NegativeOne:
+        if self.is_ZeroMatrix:
+            return self
```

Comment:
This seems to return the zero matrix for all powers while line 467 returns the identity for the 0th power of the zero matrix.